src/pytorch_lightning/core/saving.py

```diff
@@ -57,13 +60,13 @@ class ModelIO:
 
     @classmethod
     def load_from_checkpoint(
-        cls,
+        cls: Union[Type["ModelIO"], Type[LM], Type[LDM]],
```

Comment:
```suggestion
        cls: Union[Type[LM], Type[LDM]],
```
why do we need `ModelIO`?

Reply:
I think because ModelIO is a mixin and does not have a base class LM or LDM. If I remove the Type["ModelIO"] (which was already there), mypy complains:

> The erased type of self "Union[Type[LM`-1], Type[LDM`-2]]" is not a supertype of its class "Type[pytorch_lightning.core.saving.ModelIO]"  [misc]